src/tribler/core/logger/tests/test_logger.py

```diff
@@ -67,30 +56,14 @@ def test_setup_logging(logger: Mock, dict_config: Mock, yaml_safe_load: Mock):
     )
 
 
-@patch('tribler.core.logger.logger.logger')
-@patch('sys.stdout')
-@patch('sys.stderr')
-@patch('builtins.print')
-@patch('logging.basicConfig')
-def test_setup_logging_exception(basic_config: Mock, print_: Mock, stderr: Mock, stdout: Mock, logger: Mock):
-    error = ZeroDivisionError()
+@patch('tribler.core.logger.logger.format_error_description')
+def test_setup_logging_exception(mocked_format_error_description: Mock, tmp_path: Path):
+    log_dir = tmp_path
+    config_path = tmp_path / 'config.conf'
+    config_path.write_text('wrong config content')
+    setup_logging('', log_dir, config_path)
 
-    log_dir = MagicMock()
-    log_dir.__str__.return_value = '<log-dir>'
-    log_dir.exists.return_value = True
-    log_dir.joinpath.side_effect = error
-
-    config_path = MagicMock()
-    config_path.__str__.return_value = '<config-path>'
-    config_path.exists.return_value = True
-
-    setup_logging('<app-mode>', log_dir, config_path)
-
-    logger.info.assert_called_once_with(
-        "Load logger config: app_mode=<app-mode>, " "config_path=<config-path>, dir=<log-dir>"
-    )
-    print_.assert_called_once_with('Error in loading logger config. Using default configs. ', 'ZeroDivisionError: ', file=stderr)
-    basic_config.assert_called_once_with(level=logging.INFO, stream=stdout)
+    assert mocked_format_error_description.called
```

Comment:
Same here, it is not obvious why the test for `format_error_description` call is necessary and enough (now and in the future). It looks more safe to test wether `basicConfig(...)/dictConfig(...)` were called, something like this:

```python
@patch('tribler.core.logger.logger.logger')
@patch('builtins.print')
@patch('logging.basicConfig')
@patch('logging.config.dictConfig')
def test_setup_logging_exception(dict_config: Mock, basic_config: Mock, print_: Mock, logger: Mock):
    error = ZeroDivisionError()

    log_dir = MagicMock()
    log_dir.__str__.return_value = '<log-dir>'
    log_dir.exists.return_value = True
    log_dir.joinpath.side_effect = error

    config_path = MagicMock()
    config_path.__str__.return_value = '<config-path>'
    config_path.exists.return_value = True

    setup_logging('<app-mode>', log_dir, config_path)

    logger.info.assert_called_once_with(
        "Load logger config: app_mode=<app-mode>, " "config_path=<config-path>, dir=<log-dir>"
    )
    print_.assert_called_once_with('Error in loading logger config. Using default configs. ', 'ZeroDivisionError: ',
                                   file=sys.stderr)
    basic_config.assert_called_once_with(level=logging.INFO, stream=sys.stdout)
    dict_config.assert_not_called()
```

Reply:
Ideally, the test should test two different things:
* that the code reaches the correct final state or performs the expected actions;
* that all code paths are covered.

In the case of  `test_setup_logging_exception`, checking just the call of `format_error_description` is intended to test that the code visited a specific code path. But the test does not check the desired result, namely that  `logging.basicConfig(...)` was called as a result of the action. It is possible that later, after some new refactoring, `tribler.core.logger.logger.format_error_description` will still be called, but `logging.basicConfig(...)` will not be called anymore. The test that only intends to achieve coverage does not check this. It makes the codebase more brittle, as it can stop performing the expected actions while still passing tests.

Reply:
I've added comments and checks on the `logging.basicConfig(...)` call.